src/Service.Host/client/src/components/VatCode.js

```diff
@@ -79,6 +83,10 @@ function VatCode({
         setVatCode({ ...vatCode, [propertyName]: newValue });
     }
 
+    function vatCodeMatchesItem() {
+        return JSON.stringify(vatCode) === JSON.stringify(item);
```

Comment:
is there a neater way to test object equality?

Reply:
In this circumstance I thought that this would be the simplest. I did start at writing a function to deeply compare objects but it ended up being overkill for this I think. There isn't really a JS object comparitor without writing one yourself or pulling in a library.

The other option I tried here was:

`    function sernosSequenceMatchesItem() {      
        return Object.entries(sernosSequence).some(([key, value]) => item[key] !== value);
    }
`

This has the advantage of not comparing the order of the objects parameters but will always return false if one of those parameters is itself an object. (i.e. `item = { a: 'a', b: {z: 'z'}}`). I thought this seemed the most straightforward but happy to look at other options.

Otherwise there is directly comparing the values like

`    function sernosSequenceMatchesItem() {      
        return item.code === sernosSequence.code && item.description === sernosSequence.description etc etc;
    }
`

But that seemed horrible

Reply:
Why do you have to do it. 
I just set the mode to editing as soon as they make a change. Does that cause a problem that I haven't noticed or are you just being extra kind to the user?

Reply:
Looking into it more, I think this is a problem with how we've been using edit status. Our components don't have a way to be both 'creating' and 'editing', so there is a problem with code like:

`
                        <SaveBackCancelButtons
                            saveDisabled={
                                !this.editing() ||
                                (!saCoreType.coreType || saCoreType.coreType.length === 0)
                            }
                            saveClick={this.creating() ? this.handleAddClick : this.handleSaveClick}
                            cancelClick={this.handleCancelClick}
                            backClick={this.handleBackClick}
                        />
`

(this ones in SaCoreTypes just to use a different example)

this.handleAddClick will never be called as currently we always set editStatus to 'edit' when a field is updated. this.creating() will never evaluate to true when the save button is active.

The code in this PR works because we don't send an action on creating an item. Therefore we don't add an empty item into the redux store and editStatus will always be 'create' as passed in by the container. (I actually just realised this morning that this was what was actually happening...)

I think we'd need a comparison like this as we want the save button to be disabled if the editStatus is 'viewing', the input is invalid or edit status is 'edit' but the fields have not been changed. (the user has edited a field then put it back to its original value) . I think this would require a check on the local state and the item from props. 

So actually I think the correct solution might be to fire off a new action when creating an item and not update editStatus to 'edit' when a field is changed if we are creating a new item.

I've waffled on a bit here and this might all be nonsense so let me know if I'm just over complicating things and have missed something obvious...

Reply:
I now dispatch `REQUEST_CREATE_{ITEM}` from the create container and I've updated the relevant reducer to clear the item.